consumer/protocol/rpc_extensions_test.go

```diff
@@ -127,6 +130,8 @@ func (s *RPCSuite) TestApplyRequestValidationCases(c *gc.C) {
 	req.Changes[1].Delete = "other-valid-id"
 	c.Check(req.Validate(), gc.ErrorMatches, `Changes\[1\]: invalid ExpectModRevision \(0; expected > 0 or -1\)`)
 	req.Changes[1].ExpectModRevision = 1
+	c.Check(req.Validate(), gc.ErrorMatches, `Changes\[1\]: hints may be set only with an upsert, not a delete`)
```

Comment:
I'd expect to see this same validation error for the change at index 0, since the validation checks whether both `Delete` and `PrimaryHints` are set.

Reply:
delete and upsert raises a different error, and test code above clears the `delete` field so it's unset at this point.